cpp/fetcher/remote_peer.cc

```diff
@@ -16,29 +19,46 @@ using util::TaskHold;
 
 namespace cert_trans {
 
+DEFINE_int32(remote_peer_sth_refresh_interval_seconds, 10,
+             "Number of seconds between checks for updated STHs from the "
+             "remote peer.");
+
 
 struct RemotePeer::Impl {
-  Impl(unique_ptr<LogVerifier>&& verifier,
+  Impl(unique_ptr<LogVerifier>&& verifier, AsyncLogClient* client,
        const std::function<void(const ct::SignedTreeHead&)>& on_new_sth,
        util::Task* task)
       : verifier_(move(verifier)),
+        client_(CHECK_NOTNULL(client)),
         on_new_sth_(on_new_sth),
         task_(CHECK_NOTNULL(task)) {
     CHECK(verifier_);
   }
 
   const std::unique_ptr<LogVerifier> verifier_;
+  AsyncLogClient* const client_;
   const std::function<void(const ct::SignedTreeHead&)> on_new_sth_;
   util::Task* const task_;
 
   mutex lock_;
   shared_ptr<SignedTreeHead> sth_;
 
+  void FetchSTH(util::Task* task);
   void DoneGetSTH(const std::shared_ptr<ct::SignedTreeHead>& on_new_sth,
                   AsyncLogClient::Status status);
 };
 
 
+void RemotePeer::Impl::FetchSTH(util::Task* task) {
+  if (CHECK_NOTNULL(task)->CancelRequested()) {
+    task->Return(util::Status::CANCELLED);
```

Comment:
Wrong task here, for this condition and this `Return()`, you want to use `task_`, I think. `CancelRequested()` is a bit odd (since `task` is an already completed task!), but might work. The `Return()` has to be on `task_`, for sure, though! :smiley: